eng/pipelines/templates/steps/build-package-artifacts.yml

```
@@ -76,6 +76,11 @@ steps:
         Write-Host "We freshly generated the PR diff, and this is what it sees!"
         Get-ChildItem -R -Force $(Build.ArtifactStagingDirectory)/diff | % { $_.FullName }
         cat $(Build.ArtifactStagingDirectory)/diff/diff.json
+
+        $content = Get-Content -Raw $(Build.ArtifactStagingDirectory)/diff/diff.json | ConvertFrom-Json
+        foreach ($item in $content.ChangedServices) {
+          Write-Host "##vso[build.addbuildtag]$item"
+        }
```

Pipeline Witness can be updated to process the AdditionalTags attachment.

```suggestion
        if ($content.ChangedServices) {
          Write-Host "Attaching changed service names to the build for additional tag generation"
          $content.ChangedServices | ConvertTo-Json | Out-File -FilePath $(System.DefaultWorkingDirectory)/tags.json -Encoding utf8
          Write-Host '##vso[task.addattachment type=AdditionalTags;name=AdditionalTags;]$(System.DefaultWorkingDirectory)/tags.json'
        }
```

---

I'll PR this change over to `save-package-properties.yml` so that everyone can just get this for free.